src/test/java/com/exasol/adapter/dialects/hive/HiveSqlDialectIT.java

```diff
@@ -1,385 +1,652 @@
+        createVirtualSchema(VIRTUAL_SCHEMA_HIVE_JDBC_NUMBER_TO_DECIMAL,
+                Optional.of("hive_cast_number_to_decimal_with_precision_and_scale='36,2'"));
     }
 
-    @Test
-    public void testSetup() throws SQLException {
-        final String query = "SELECT X FROM " + VIRTUAL_SCHEMA + ".T";
-        final ResultSet result = executeQuery(query);
-        assertNextRow(result, Long.valueOf("99"));
+    private static Connection getHiveConnection() throws ClassNotFoundException, SQLException, InterruptedException {
+        Class.forName("org.apache.hive.jdbc.HiveDriver");
+        return DriverManager.getConnection("jdbc:hive2://localhost:" + HIVE_EXPOSED_PORT + "/" + SCHEMA_HIVE,
+                HIVE_USERNAME, HIVE_PASSWORD);
     }
 
-    // Join Tests -------------------------------------------------------------
-    @Test
-    void testInnerJoin() throws SQLException {
-        final String query = "SELECT * FROM  " + VIRTUAL_SCHEMA + ".t1 a INNER JOIN  " + VIRTUAL_SCHEMA
-                + ".t2 b ON a.x=b.x";
-        final ResultSet result = executeQuery(query);
-        assertAll(() -> assertNextRow(result, 2L, "bbb", 2L, "bbb"), () -> assertFalse(result.next()));
+    private static String getPropertyFromFile(final String filePath, final String propertyName) {
+        try (final InputStream inputStream = new FileInputStream(filePath)) {
+            final Properties properties = new Properties();
+            properties.load(inputStream);
+            return properties.getProperty(propertyName);
+        } catch (final IOException exception) {
+            throw new IllegalArgumentException("Cannot access the oracle.properties file or read from it.");
+        }
     }
 
-    @Test
-    void testInnerJoinWithProjection() throws SQLException {
-        final String query = "SELECT b.y || " + VIRTUAL_SCHEMA + ".t1.y FROM  " + VIRTUAL_SCHEMA + ".t1 INNER JOIN  "
-                + VIRTUAL_SCHEMA + ".t2 b ON " + VIRTUAL_SCHEMA + ".t1.x=b.x";
-        final ResultSet result = executeQuery(query);
-        assertAll(() -> assertNextRow(result, "bbbbbb"), () -> assertFalse(result.next()));
+    private static void uploadFilesToBucket(final String driverName, final String driverPath)
+            throws InterruptedException, BucketAccessException, TimeoutException {
+        final Bucket bucket = exasolContainer.getDefaultBucket();
+        bucket.uploadFile(PATH_TO_VIRTUAL_SCHEMAS_JAR, VIRTUAL_SCHEMAS_JAR_NAME_AND_VERSION);
+        bucket.uploadFile(Path.of(driverPath, driverName), "drivers/jdbc/" + driverName);
+        bucket.uploadFile(HIVE_DRIVER_SETTINGS_PATH, "drivers/jdbc/" + JDBC_DRIVER_CONFIGURATION_FILE_NAME);
+    }
+
+    private static void createTableHiveSimple(final Statement statementHive) throws SQLException {
+        statementHive.execute("CREATE TABLE " + TABLE_HIVE_SIMPLE + "(X INT)");
+        statementHive.execute("TRUNCATE TABLE " + TABLE_HIVE_SIMPLE);
```

Comment:
Why do you truncate immediately after creating the table?

Reply:
I have no idea, it's legacy code